### ISSUE TITLE
Unable to create new users within app

### ISSUE THREAD

**Issue by joebudi** [Budibase/budibase#921]
**Describe the bug**
I wanted my power users to have the ability to view all the app's users, edit the users' details, and create new users. I was able to add the screens and bind to the backend, but unable to add a new user. 

**To Reproduce**
1. Create new screens (choose the user list template, user detail, user new) under the Power-user role
2. Preview app
3. Attempt to add a new user
4. Notice how the new user is not added

**Expected behavior**
New user should be added

**Screenshots**
https://www.loom.com/share/c11d1ab0f0d84486a49927203c1350d5

**Desktop (please complete the following information):**
Mac
Chrome

**Additional context**

**Comment by aptkingston** [comment]
Kev raised this issue in the comments of the security PR - basically we can't do this right now. Users are not just a normal table and so they don't work with the usual front end components. If we want to be able to add users within an app then we'll need to build some custom components to handle it.

Technical bit:
The reason it is failing is because there is no password. The password isn't in the user table schema deliberately. The roles there are also actually ID's rather than the pretty names - it just so happens that those built in roles have ID's which are the same as their pretty names in caps. If you add a custom role then you'll see an ugly long ID when creating a user.

Basically listing / editing / creating users is something we don't support right now, and I might remove the users table from being a valid data source so that you can't link it up to screens at all.

**Comment by aptkingston** [comment]
I've thrown up a PR here https://github.com/Budibase/budibase/pull/924 to remove the users table from being an option when choosing a table or view data source in the front end.
